covid_model/ode_builder.py

```diff
@@ -155,35 +155,19 @@ def __init__(self, base_ode_builder=None, trange=None, attributes: OrderedDict =
         self.solution_ydf = None
 
         if base_ode_builder is not None:
-            self.params = copy.deepcopy(base_ode_builder.params)
+            if deepcopy_params:
+                self.params = {t: {cmpt: params.copy() for cmpt, params in params_by_cmpt.items()} for t, params_by_cmpt in base_ode_builder.params.items() for t in self.trange}
+            else:
+                self.params = {t: params_by_cmpt for t, params_by_cmpt in base_ode_builder.params.items() for t in self.trange}
             self.terms = {term.deepcopy() for term in base_ode_builder.terms}
-            self.linear_matrix = {t: copy.deepcopy(m) for t, m in base_ode_builder.linear_matrix.items() if t in self.trange}
-            self.nonlinear_matrices = {t: copy.deepcopy(m) for t, m in base_ode_builder.nonlinear_matrices.items() if t in self.trange}
-            self.constant_vector = {t: copy.deepcopy(m) for t, m in base_ode_builder.constant_vector.items() if t in self.trange}
-            self.nonlinear_multiplier = copy.deepcopy(base_ode_builder.nonlinear_multiplier)
+            self.linear_matrix = {t: m.copy() for t, m in base_ode_builder.linear_matrix.items() if t in self.trange}
+            self.nonlinear_matrices = {t: m.copy() for t, m in base_ode_builder.nonlinear_matrices.items() if t in self.trange}
+            self.constant_vector = {t: m.copy() for t, m in base_ode_builder.constant_vector.items() if t in self.trange}
```

Comment:
These are scipy sparse matrices, which have their own copy command; no need to copy.deepcopy(), which can be extremely slow.